io/plugins/it.geosolutions.hale.io.appschema/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingGenerator.java

```diff
@@ -526,10 +527,36 @@ static QName tryInferNamespacePrefix(PropertyDefinition child) {
 						prefix = defaultType.getName().getPrefix();
 				}
 			}
+			else {
+				String newPrefix = getPrefix(dpd);
+				if (StringUtils.isNotBlank(newPrefix)) {
+					prefix = newPrefix;
+				}
+			}
 		}
 		return new QName(namespaceURI, "", prefix);
 	}
 
+	private static String getPrefix(PropertyDefinition dpd) {
+		try {
+			if (dpd instanceof AbstractPropertyDecorator) {
+				AbstractPropertyDecorator cdp = (AbstractPropertyDecorator) dpd;
+				PropertyDefinition property = cdp.getDecoratedProperty();
+				if (property instanceof AbstractPropertyDecorator) {
+					DefinitionGroup declaringGroup = ((AbstractPropertyDecorator) property)
+							.getDeclaringGroup();
+					if (declaringGroup instanceof XmlTypeDefinition) {
+						XmlTypeDefinition typeDef = (XmlTypeDefinition) declaringGroup;
+						return typeDef.getName().getPrefix();
+					}
+				}
+			}
+		} catch (NullPointerException ex) {
+			log.warn("Prefix not found", ex);
+		}
+		return null;
```

Comment:
This doesn't look right, we need to provide good information to the user:
- where can the null pointer exception happen? use an ``if`` and put a decent log on it
- returning ``NULL`` doesn't look like a normal returning value for me